### PR TITLE
Fix hash files function

### PR DESCRIPTION
**Pull request by YevheniiSemendiak**
Fixed problems:
- we skiped all remaining files matched file pattern after first empty file occuracne ([click](https://github.com/neuro-inc/neuro-flow/blob/master/neuro_flow/expr.py#L197))
- we didn't compute the cache of entire files, only first 256 kBytes

I've also added `--version` option to CLI main group, if you don't mind. It helps sometimes to debug the problems, especially with clients.

We could also include filenames into cache calculation with smth like
```
...
for fname in sorted(workspace.glob(pattern)):
    hasher.update(bytes(str(fname), "UTF-8"))
    ...
```
wdyt, @asvetlov ?

### CODE REVIEW

**Comment by asvetlov** [comment]
Good catch!
Please go ahead.
Adding filenames to hash also sounds pretty good; it allows to catch, e.g., a configuration file renaming.

A small suggestion: instead of `stream.read(bufsize)` please use https://docs.python.org/3/library/io.html#io.BufferedIOBase.readinto1 for the sake of performance.

The single `bytearray` buffer per the whole call is needed, the size should be in the order of megabytes (by default socket has 4-6 MB for both RECV and SEND buffers); I guess 16MB for our needs is not too high.

**Comment by YevheniiSemendiak** [comment]
@asvetlov,
Thanks for the advice,
I've added a single-per-call buffer and file name hash, but it behaves differently on Win+Python3.6 ([run](https://github.com/neuro-inc/neuro-flow/runs/1381003638?check_suite_focus=true#step:7:38)), while on other setups it is OK. Could you give a hint, what might be wrong here? 

Also, I have a doubt that casting [like this](https://github.com/neuro-inc/neuro-flow/pull/190/commits/ca21ef91a4c2438fb1c6680bfbdd91f30dda30cd) to fix a linter is correct (yet, it works OK), wdyt?

File: neuro_flow/expr.py [hunk 1]
```diff
@@ -191,11 +193,13 @@ async def hash_files(ctx: CallCtx, *patterns: str) -> str:
...
         for fname in sorted(workspace.glob(pattern)):
-            with fname.open("rb") as stream:
-                data = stream.read(256 * 1024)
-                if not data:
-                    break
-                hasher.update(data)
+            relative_fname = fname.relative_to(workspace).as_posix()
+            hasher.update(bytes(relative_fname, "UTF-8"))
+            with fname.open("rb", buffering=0) as stream:
+                read = stream.readinto(buffer)  # type: ignore
```
Comment (asvetlov) [inline neuro_flow/expr.py:199]: typeshed has incomplete annotations for pathlib; the master seems fixed.

File: neuro_flow/expr.py [hunk 2]
```diff
@@ -191,11 +193,15 @@ async def hash_files(ctx: CallCtx, *patterns: str) -> str:
     )
     for pattern in patterns:
         for fname in sorted(workspace.glob(pattern)):
-            with fname.open("rb") as stream:
-                data = stream.read(256 * 1024)
-                if not data:
-                    break
-                hasher.update(data)
+            # On Windows the Python 3.6 glob() returns lower-cased filenames,
```
Comment (asvetlov) [inline neuro_flow/expr.py:196]: This was the reason for the CI failure

Comment (YevheniiSemendiak) [reply neuro_flow/expr.py:196]: Ok, thanks for the hint!

**Pull request merged**